proposals/0000-reduce-fragmentation.md

```diff
@@ -0,0 +1,837 @@
+
+-----
+
+# Environment API
+
+## document
+
+Support the `document` object for common patterns such as listening to document resize, listening to capture or bubble phase events for the entire application (e.g., "outside click" pattern), etc.
+
+### Properties
+
+- [ ] [`activeElement`](https://developer.mozilla.org/en-US/docs/Web/API/Document/activeElement).
+- [ ] [`visibilityState`](https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilityState)
+
+### Events
+
+* [ ] `scroll`.
+* [ ] `visibilitychange`.
+
+## IntersectionObserver
+
+Support the [`IntersectionObserver`](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver) API for observing the intersection of target elements. This can also be used as a building block for performance tooling.
+
+## MutationObserver
+
+Support the latest [`MutationObserver`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) API for watching changes to the host node tree. This can also be used as a building block for performance tooling.
+
+## ResizeObserver
+
+Support the latest [`ResizeObserver`](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) API for responding the changes in the size of elements. Note that this API no longer includes positional coordinates and is optimized for width/height information.
+
+-----
+
+# Components API (DOM subset)
+
+A logical extension of support more React DOM props is to support a subset of React DOM components directly.
+
+This would provide a built-in declarative API for cross-platform elements in React Native. All unknown tags default to `<span>` layout. All known-but-unimplemented block-level tags default to `<div>` layout. Most tags can be mapped to existing React Native components and props, in the reverse of the [mapping used by React Native for Web](https://github.com/necolas/react-native-web/blob/b2eb3ca03ffc4b8570591b6c2082c6e4edae0e7f/packages/react-native-web/src/modules/AccessibilityUtil/propsToAccessibilityComponent.js#L12-L30).
+
+## Sections
+
+Content sectioning elements allow you to organize the document content into logical pieces. Use the sectioning elements to create a broad outline for your page content, including header and footer navigation, and heading elements to identify sections of content.
+
+* [ ] `article` is equivalent to `View` with `display:block` layout and `role="article"`.
+* [ ] `aside`
+* [ ] `div` is equivalent to `View` with `display:block` layout.
+* [ ] `footer`.
+* [ ] `h1-6`.
+* [ ] `header`.
+* [ ] `main`.
+* [ ] `nav`.
+* [ ] `section` is equivalent to `View` with `display:block` layout and `role="section"`.
+
+## Text
+
+Text content elements organize blocks or sections of content. Important for accessibility, these elements identify the purpose or structure of that content.
+
+* [ ] `blockquote`.
+* [ ] `li`.
+* [ ] `ol`.
+* [ ] `ul`.
```

Comment:
RN should probably render these with no styles by default. On the web side, we'll reset the styles as is typical for web app development.

Reply:
Interesting… I had never considered the necessity for web compliance to also involve either implementing default styles or including a reset stylesheet. (It's been so long since I've had to set one of those up…)

I *think* I agree with assuming common reset practices…